src/main/java/ch/njol/skript/expressions/ExprMaxPlayers.java

```diff
@@ -18,107 +18,135 @@
-@Since("2.3")
-public class ExprMaxPlayers extends SimpleExpression<Long> {
+@RequiredPlugins("Paper 1.16+ (modify max real players)")
+@Since("2.3, INSERT VERSION (modify max real players)")
+public class ExprMaxPlayers extends SimpleExpression<Integer> {
 
 	static {
-		Skript.registerExpression(ExprMaxPlayers.class, Long.class, ExpressionType.PROPERTY,
+		Skript.registerExpression(ExprMaxPlayers.class, Integer.class, ExpressionType.PROPERTY,
 				"[the] [(1¦(real|default)|2¦(fake|shown|displayed))] max[imum] player[s] [(count|amount|number|size)]",
 				"[the] [(1¦(real|default)|2¦(fake|shown|displayed))] max[imum] (count|amount|number|size) of players");
 	}
 
 	private static final boolean PAPER_EVENT_EXISTS = Skript.classExists("com.destroystokyo.paper.event.server.PaperServerListPingEvent");
+	private static final boolean SET_MAX_PLAYERS_EXISTS = Skript.methodExists(Server.class, "setMaxPlayers", int.class);
 
 	private boolean isReal;
 
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
 		boolean isServerPingEvent = getParser().isCurrentEvent(ServerListPingEvent.class) ||
 				(PAPER_EVENT_EXISTS && getParser().isCurrentEvent(PaperServerListPingEvent.class));
+		
 		if (parseResult.mark == 2 && !isServerPingEvent) {
 			Skript.error("The 'shown' max players count expression can't be used outside of a server list ping event");
 			return false;
 		}
+		
 		isReal = (parseResult.mark == 0 && !isServerPingEvent) || parseResult.mark == 1;
 		return true;
 	}
 
 	@Override
 	@Nullable
-	public Long[] get(Event e) {
-		if (!isReal && !(e instanceof ServerListPingEvent))
+	public Integer[] get(Event event) {
+		if (!isReal && !(event instanceof ServerListPingEvent))
 			return null;
 
-		if (isReal)
-			return CollectionUtils.array((long) Bukkit.getMaxPlayers());
-		else
-			return CollectionUtils.array((long) ((ServerListPingEvent) e).getMaxPlayers());
+		if (isReal) {
+			return CollectionUtils.array(Bukkit.getMaxPlayers());
+		} else {
+			return CollectionUtils.array(((ServerListPingEvent) event).getMaxPlayers());
+		}
 	}
 
 	@Override
 	@Nullable
 	public Class<?>[] acceptChange(ChangeMode mode) {
-		if (!isReal) {
-			if (getParser().getHasDelayBefore().isTrue()) {
-				Skript.error("Can't change the fake max players count anymore after the server list ping event has already passed");
-				return null;
-			}
-			switch (mode) {
-				case SET:
-				case ADD:
-				case REMOVE:
-				case DELETE:
-				case RESET:
-					return CollectionUtils.array(Number.class);
-			}
+		if (!isReal && getParser().getHasDelayBefore().isTrue()) {
+			Skript.error("Can't change the fake max players count anymore after the server list ping event has already passed");
+			return null;
+		}
+		
+		if (isReal && !SET_MAX_PLAYERS_EXISTS) {
+			Skript.error("Modifying real max player count is not supported on this server version");
```

Comment:
```suggestion
			Skript.error("Modifying the 'real max player count' is only supported on Paper 1.16 and newer");
```
Might be more helpful for users